asapdiscovery-alchemy/asapdiscovery/alchemy/cli/prep.py

```diff
@@ -0,0 +1,170 @@
+    help="The name of the JSON file containing the AlchemyPrepWorkflow factory, if not supplied the default will be used.",
+)
+@click.option(
+    "-n",
+    "--dataset-name",
+    type=click.STRING,
+    help="The name of the AlchemyDataset this will also be the name of the folder created.",
+)
+@click.option(
+    "-l",
+    "--ligands",
+    type=click.Path(resolve_path=True, exists=True, file_okay=True, dir_okay=False),
+    help="The file which contains the ligands to use in the planned network.",
+)
+@click.option(
+    "-r",
+    "--receptor-complex",
+    type=click.Path(resolve_path=True, exists=True, file_okay=True, dir_okay=False),
+    help="The name of the file which contains the prepared receptor complex including the crystal ligand.",
+)
+@click.option(
+    "-cs",
+    "--core-smarts",
+    type=click.STRING,
+    help="The SMARTS string used to identify the atoms in each ligand which should be constrained to the reference pose.",
+)
+def run(
+    dataset_name: str,
+    ligands: str,
+    receptor_complex: str,
+    factory_file: Optional[str] = None,
+    core_smarts: Optional[str] = None,
+):
+    """
+    Create an AlchemyDataset by running the given AlchemyPrepWorkflow which will expand the ligand states and generate
+    constrained poses suitable for ASAP-Alchemy.
+
+    Parameters
+    ----------
+    dataset_name: The name which should be given to the AlchemyDataset all results will be saved in a folder with the same name.
+    ligands: The name of the local file which contains the input ligands to be prepared in the workflow.
+    receptor_complex: The name of the local file which contains the prepared complex including the crystal ligand.
+    factory_file: The name of the JSON file with the configured AlchemyPrepWorkflow, if not supplied the default will be
+        used but a core smarts must be provided.
+    core_smarts: The SMARTS string used to identify the atoms in each ligand to be constrained. Required if the factory file is not supplied.
+    """
+    from asapdiscovery.data.openeye import save_openeye_sdfs
+    from asapdiscovery.data.schema_v2.complex import PreppedComplex
+    from asapdiscovery.data.schema_v2.molfile import MolFileFactory
+
+    pretty.install()
+    console = rich.get_console()
+    print_header(console)
+
+    # load the factory and set the core smarts if supplied
+    if factory_file is not None:
+        factory = AlchemyPrepWorkflow.parse_file(factory_file)
+        if core_smarts is not None:
+            factory.core_smarts = core_smarts
+    else:
+        factory = AlchemyPrepWorkflow(core_smarts=core_smarts)
+
+    # load the molecules
+    asap_ligands = MolFileFactory.from_file(filename=ligands).ligands
+
+    message = Padding(
+        f"Loaded {len(asap_ligands)} ligands from [repr.filename]{ligands}[/repr.filename]",
+        (1, 0, 1, 0),
+    )
+    console.print(message)
+
+    # TODO make a general from_file which handles this
+    if ".json" in receptor_complex:
```

Comment:
Should this also be able to take a PDB and prep it de-novo?

Reply:
I think I want to ensure the same prepared complex is used for free energy calcs so for now ill insist that its prepped before hand.